Cargo.toml

```diff
@@ -82,7 +82,7 @@ libp2p-core = { version = "0.43.1", path = "core" }
 libp2p-dcutr = { version = "0.14.0", path = "protocols/dcutr" }
 libp2p-dns = { version = "0.44.0", path = "transports/dns" }
 libp2p-floodsub = { version = "0.47.0", path = "protocols/floodsub" }
-libp2p-gossipsub = { version = "0.50.0", path = "protocols/gossipsub" }
+libp2p-gossipsub = { version = "0.51.0", path = "protocols/gossipsub" }
```

Comment:
`0.50.0` isn't published yet, so no need for a version bump. Same below for the CHANGELOG entry,